test/e2e/common/setup.go

```diff
@@ -0,0 +1,310 @@
+		if _, err := cluster.Kube().CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "istio-system",
+			},
+		}, v1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create namespace: %w", err)
+		}
+		return nil
+	}
+
+	for _, c := range ctx.Clusters() {
+		if err := retry.UntilSuccess(func() error {
+			_, err := c.Kube().CoreV1().Namespaces().Get(context.Background(), "istio-system", v1.GetOptions{})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("failed to get namespace: %w", err)
+				}
+				return createNamespace(c)
+			}
+			if err := c.Kube().CoreV1().Namespaces().Delete(context.Background(), "istio-system", v1.DeleteOptions{}); err != nil {
+				return fmt.Errorf("failed to delete namespace: %w", err)
+			}
+			return createNamespace(c)
+		}, retry.Timeout(30*time.Second), retry.Delay(1*time.Second)); err != nil {
+			return err
+		}
+	}
+
+	ctx.Cleanup(func() {
+		for idx, c := range ctx.Clusters() {
+			if err := c.Kube().CoreV1().Namespaces().Delete(context.Background(), "istio-system", v1.DeleteOptions{}); err != nil {
+				scopes.Framework.Errorf("failed to delete control plane namespace (cluster=%s): %w", clusterNames[idx], err)
+			}
+		}
+	})
+	return nil
+}
+
+func CreateCACertsSecret(ctx resource.Context) error {
+	for idx, c := range ctx.Clusters() {
+		clusterName := clusterNames[idx]
+		data := map[string][]byte{
+			"root-cert.pem":  {},
+			"cert-chain.pem": {},
+			"ca-cert.pem":    {},
+			"ca-key.pem":     {},
+		}
+		if err := setCacertKeys(fmt.Sprintf("%s/test/testdata/certs/%s", rootDir, clusterName), data); err != nil {
+			return fmt.Errorf("failed to set keys in cacerts secret (cluster=%s): %w", clusterName, err)
+		}
+		cacerts := &corev1.Secret{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "cacerts",
+				Namespace: "istio-system",
+			},
+			Data: data,
+		}
+		if _, err := c.Kube().CoreV1().Secrets("istio-system").Create(context.Background(), cacerts, v1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create cacerts secret (cluster=%s): %w", clusterName, err)
+		}
+	}
+	return nil
+}
+
+func setCacertKeys(dir string, data map[string][]byte) error {
+	for key := range data {
+		fileName := fmt.Sprintf("%s/%s", dir, key)
+		fileData, err := os.ReadFile(fileName)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", fileName, err)
+		}
+		data[key] = fileData
+	}
+	return nil
+}
+
+// DeployControlPlanes deploys Istio using the manifest generated from IstioOperator resource.
+// We can't utilize standard Istio installation supported by the Istio framework,
+// because it does not allow to apply different Istio settings to different primary clusters
+// and always sets up direct access to the k8s api-server, while it's not desired in mesh federation.
+func DeployControlPlanes(federationControllerConfigMode string) resource.SetupFn {
```

Comment:
A narrower `ConfigMode` type and some constants have been added here: https://github.com/openshift-service-mesh/federation/blob/master/internal/pkg/config/federation_config.go#L109-L114

I think we should use this type in function signatures and use the constants where the functions are invoked.
```suggestion
func DeployControlPlanes(federationControllerConfigMode config.ConfigMode) resource.SetupFn {
```